elastic/kibana
# [Fleet] Provide license info in telemetry

### Summary

Relates https://github.com/elastic/ingest-dev/issues/2866

This PR adds license information to Fleet telemetry events stored in the `fleet-usages*` index. The `license. issued_to` field provides the customer name.

### Testing locally

In a local setup, only the new `license_issued_to` field is defined.

Fleet usage logger:
1. Change the [interval](https://github.com/elastic/kibana/blob/de755234957b8df5bb773587150014b076fe5551/x-pack/plugins/fleet/server/services/fleet_usage_logger.ts#L59C46-L59C48) of the `FleetUsageLogger` to e.g. `1m`.
2. Wait for the log line to show up:
```
Fleet Usage: {"agents_enabled":true,"agents":{"total_enrolled":3,"healthy":3,"unhealthy":0,"offline":0,"inactive":0,"unenrolled":0,"total_all_statuses":3,"updating":0},"fleet_server":{"total_enrolled":1,"healthy":1,"unhealthy":0,"offline":0,"updating":0,"total_all_statuses":1,"num_host_urls":1},"license_issued_to":"elasticsearch"}
```

Fleet usage sender:
1. Change the interval of the `FleetUsageSender` to e.g. `1m`.
2. Wait for the following log line:
```
[2024-05-03T15:24:11.843+02:00][DEBUG][plugins.fleet] Fleet usage telemetry: {"agents_enabled":true,"agents":{"total_enrolled":3,"healthy":3,"unhealthy":0,"offline":0,"inactive":0,"unenrolled":0,"total_all_statuses":3,"updating":0},"fleet_server":{"total_enrolled":1,"healthy":1,"unhealthy":0,"offline":0,"updating":0,"total_all_statuses":1,"num_host_urls":1},"packages":[{"name":"system","version":"1.55.2","enabled":true},{"name":"synthetics","version":"1.2.1","enabled":false},{"name":"fleet_server","version":"1.5.0","enabled":true},{"name":"elastic_agent","version":"1.18.0","enabled":false},{"name":"nginx","version":"1.20.0","enabled":false}],"agent_checkin_status":{"error":0,"degraded":0},"agents_per_policy":[2,1],"agents_per_os":[{"name":"Ubuntu","version":"20.04.6 LTS (Focal Fossa)","count":3}],"fleet_server_config":{"policies":[{"input_config":{}}]},"agent_policies":{"count":3,"output_types":["elasticsearch"]},"agent_logs_panics_last_hour":[],"agent_logs_top_errors":[],"fleet_server_logs_top_errors":[],"license_issued_to":"elasticsearch"}
```

Upgrade sender:
1. Change the [interval](https://github.com/elastic/kibana/blob/b8d8c737e6cc7889c19a6e7984d618bf378ee617/x-pack/plugins/fleet/server/services/telemetry/fleet_usage_sender.ts#L30) of the `FleetUsageSender` to e.g. `1m`.
4. (Re)install a package and wait for the telemetry log (DEBUG level), which should contain license information:
```
[2024-05-03T15:20:41.087+02:00][DEBUG][plugins.fleet.telemetry_events] [{"packageName":"nginx","currentVersion":"1.20.0","newVersion":"1.20.0","status":"success","dryRun":false,"eventType":"package-install","installType":"reinstall","errorMessage":[],"license_issued_to":"elasticsearch"}]
```

### Checklist
